### DISCUSSION TITLE
[Bug] (httpasyncclient-4.x-plugin) Span is error forever because of org.apache.skywalking.apm.plugin.httpasyncclient.v4.wrapper.HttpAsyncResponseConsumerWrapper.responseReceived>>errorOccurred()

### DISCUSSION THREAD

**Discussion by gzniuniu** [apache/skywalking#11546] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

use the elasticsearch to search,then found the error response int the "Trace", the answer is right and http status code is 200 ,but the "Error" shows  "true".why it happend because of the span is created with the errorOccurred forever in my opinion.
the code :
if (ContextManager.isActive()) {
            int statusCode = response.getStatusLine().getStatusCode();
            AbstractSpan span = ContextManager.activeSpan().errorOccurred();//this errorOccurred() lead the span error forever
            Tags.HTTP_RESPONSE_STATUS_CODE.set(span, statusCode);
            if (statusCode >= 400) {
                span.errorOccurred();
            }
            ContextManager.stopSpan();
        }

![image](https://github.com/apache/skywalking/assets/13554279/3acc019c-3b79-40f7-b142-c2a51bebe810)

### What you expected to happen

when i send receiver the right response , the "Error" should be flase

### How to reproduce

use the elasticsearch to search,then found the error response int the "Trace", but the answer is right ,but the "Error" shows  "true"

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by gzniuniu** [comment]
![image](https://github.com/apache/skywalking/assets/13554279/db4fe418-8bf5-4956-91c9-962b02576f0a)

**Comment by wu-sheng** [comment, marked as answer]
I think this has been fixed https://github.com/apache/skywalking-java/pull/608. It will be included in the 9.1.0.

**Reply by gzniuniu** [reply]
thanks
